Gems/Atom/Utils/Code/Source/ImGuiGpuProfiler.cpp

```diff
@@ -1535,6 +1684,242 @@ namespace AZ
+                {
+                    memoryType = 1;
+                }
+                else
+                {
+                    continue;
+                }
+
+                for (const auto& buffer : pool.m_buffers)
+                {
+                    line = AZStd::string::format(
+                        MemoryCSVRowFormat, pool.m_name.GetCStr(), memoryType, buffer.m_name.GetCStr(), 0, buffer.m_sizeInBytes,
+                        static_cast<uint32_t>(buffer.m_bindFlags));
+                    fileOut.Write(line.data(), line.size());
+                }
+
+                for (const auto& image : pool.m_images)
+                {
+                    line = AZStd::string::format(
+                        MemoryCSVRowFormat, pool.m_name.GetCStr(), memoryType, image.m_name.GetCStr(), 1, image.m_sizeInBytes,
+                        static_cast<uint32_t>(image.m_bindFlags));
+                    fileOut.Write(line.data(), line.size());
+                }
+            }
+        }
+
+        template <typename T>
+        bool parseCSVField(std::istringstream& stream, std::string& field, T& out, char delimiter = ',')
+        {
+            if (std::getline(stream, field, delimiter))
+            {
+                if constexpr (AZStd::is_same_v<T, int>)
+                {
+                    if (sscanf_s(field.c_str(), "%i", &out) != 1)
+                    {
+                        return false;
+                    }
+                }
+                else if constexpr (AZStd::is_same_v<T, uint32_t>)
+                {
+                    if (sscanf_s(field.c_str(), "%" PRIu32, &out) != 1)
+                    {
+                        return false;
+                    }
+                }
+                else if constexpr (AZStd::is_same_v<T, uint64_t>)
+                {
+                    if (sscanf_s(field.c_str(), "%" PRIu64, &out) != 1)
+                    {
+                        return false;
+                    }
+                }
+                else if constexpr (AZStd::is_same_v<T, AZ::Name>)
+                {
+                    out = AZ::Name{ field.c_str() };
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
+                return true;
+            }
+            return false;
+        }
+
+        void ImGuiGpuMemoryView::LoadFromCSV(const AZStd::string& fileName)
+        {
+            AZ::IO::SystemFile fileIn;
+
+            if (!fileIn.Open(fileName.c_str(), AZ::IO::SystemFile::SF_OPEN_READ_ONLY))
+            {
+                return;
+            }
+
+            std::string data;
+            data.resize(fileIn.Length());
+
+            fileIn.Read(fileIn.Length(), data.data());
+
+            std::istringstream lines{ data.c_str() };
+
+            std::string line;
+            if (!std::getline(lines, line))
```

Comment:
`std` shouldn't be used directly. If there's not an AZStd equivalent then it should be discussed and added if agreed otherwise there are commonly non-std alternatives that are preferable.

Reply:
Do we have a getline equivalent that operates on streams? I could just implement this as a forward scan I suppose

Reply:
FYI for getline specifically, I used it because there was already a precedent for it used in other tools (e.g. ImGuiManager, CrashHandler, etc.). I propose I change other uses of `std::` here to use existing facilities (e.g. `filesystem` usage) that I wasn't aware of, and for this one we defer changing it until an `AZStd::getline` is available.

Reply:
It's a little less of an issue for tools, although not recommended there either. This however is going into runtime right? Or can is it in a gem modules that can be turned off in shipping releases?

`AZ::StringFunc::TokenizeVisitor` in `\Code\Framework\AzCore\AzCore\StringFunc\StringFunc.h` might be a good candidate to replace the `std` code?